### PR TITLE
Add internal functions for throw exceptions

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - 103685

### CODE REVIEW

File: src/frontends/paddle/include/openvino/frontend/paddle/node_context.hpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ using NamedInputs = std::map<InPortName, OutputVector>;
 
 /// Keep necessary data for a single node in the original FW graph to facilitate
 /// conversion process in the rules code.
-class NodeContext : public ov::frontend::NodeContext {
+class PADDLE_API NodeContext : public ov::frontend::NodeContext {
```
Comment (rkazants) [inline src/frontends/paddle/include/openvino/frontend/paddle/node_context.hpp:22]: Do we really need to expose this externally? Ideally, think - no

Comment (rkazants) [inline src/frontends/paddle/include/openvino/frontend/paddle/node_context.hpp:22]:
```suggestion
class NodeContext : public ov::frontend::NodeContext {
```

Comment (ilya-lavrenov) [reply src/frontends/paddle/include/openvino/frontend/paddle/node_context.hpp:22]: should we move it to `src` folder then?

Comment (ilyachur) [reply src/frontends/paddle/include/openvino/frontend/paddle/node_context.hpp:22]: Yes in this case we need to remove this API from public API

File: src/core/include/openvino/core/node.hpp [hunk 2]
```diff
@@ -541,25 +541,13 @@ class OPENVINO_API NodeValidationFailure : public ov::AssertFailure {
...
-                          node->input_values().size(),
-                          " argument",
-                          (node->input_values().size() == 1 ? "" : "s"),
-                          " but got ",
-                          new_args.size());
-}
+OPENVINO_API void check_new_args_count(const Node* node, OutputVector new_args);
 
-}  // namespace ov
+[[noreturn]] OPENVINO_API void throw_on_node_validation_failure(const ov::CheckLocInfo& check_loc_info,
```
[the remark this answers is not in the document]
Comment (ilyachur) [reply src/core/include/openvino/core/node.hpp:547]: I got the difference ~ 0.4 Mb on M1 Mac.

[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/core/include/openvino/core/node.hpp:547]: M1 is not first priority platform for OV and we optimize primarily for Windows, then Linux

What I want to understand: whether noreturn attribute adds extra binary size with compare to usual function

**Review by rkazants — APPROVED** [review]
Looks good from FE perspective.

**Pull request closed without merging**